api/server/httprest/options.go

```diff
@@ -0,0 +1,34 @@
+package httprest
+
+import (
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+// Option is a http rest server functional parameter type.
+type Option func(g *Server) error
+
+// WithHTTPAddr sets the full address ( host and port ) of the server.
+func WithHTTPAddr(addr string) Option {
+	return func(g *Server) error {
+		g.cfg.httpAddr = addr
+		return nil
+	}
+}
+
+// WithRouter sets the internal router of the server, this is required.
+func WithRouter(r *mux.Router) Option {
+	return func(g *Server) error {
+		g.cfg.router = r
+		return nil
+	}
+}
+
+// WithTimeout allows changing the timeout value for API calls.
+func WithTimeout(seconds uint64) Option {
+	return func(g *Server) error {
+		g.cfg.timeout = time.Second * time.Duration(seconds)
+		return nil
+	}
+}
```

Comment:
Why is this in seconds? Why not have the argument be time.Duration? Then it can be any unit of time, not just seconds.